qiskit_machine_learning/algorithms/effective_dimension.py

```diff
@@ -0,0 +1,337 @@
+        fisher_avg = np.average(
+            np.reshape(
+                fishers,
+                (
+                    self.num_params,
+                    self.num_inputs,
+                    self._model.num_weights,
+                    self._model.num_weights,
+                ),
+            ),
+            axis=1,
+        )
+
+        # calculate f_hats for all the empirical fishers
+        f_hat = self._model.num_weights * fisher_avg / fisher_trace
+        return f_hat, fisher_trace
+
+    def _get_eff_dim(
+        self, f_hat: Union[List[float], np.ndarray], n: Union[List[int], np.ndarray, int]
+    ) -> Union[np.ndarray, int]:
+
+        if not isinstance(n, int) and len(n) > 1:
+            # expand dims for broadcasting
+            f_hat = np.expand_dims(f_hat, axis=0)
+            n_expanded = np.expand_dims(np.asarray(n), axis=(1, 2, 3))
+            logsum_axis = 1
+        else:
+            n_expanded = n # type: ignore
+            logsum_axis = None
+
+        # calculate effective dimension for each data sample size "n" out
+        # of normalized fishers
+        f_mod = f_hat * n_expanded / (2 * np.pi * np.log(n_expanded))
+        one_plus_fmod = np.eye(self._model.num_weights) + f_mod
+        # take log. of the determinant because of overflow
+        dets = np.linalg.slogdet(one_plus_fmod)[1]
+        # divide by 2 because of square root
+        dets_div = dets / 2
+        effective_dims = (
+            2
+            * (logsumexp(dets_div, axis=logsum_axis) - np.log(self.num_params))
+            / np.log(n / (2 * np.pi * np.log(n)))
+        )
+
+        return np.squeeze(effective_dims)
+
+    def get_eff_dim(self, n: Union[List[int], np.ndarray, int]) -> Union[np.ndarray, int]:
+        """
+        This method compute the effective dimension for a data sample size ``n``.
+
+        Args:
+            n: array of sample sizes
+        Returns:
+             effective_dim: array of effective dimensions for each sample size in n.
+        """
+
+        # pylint: disable=wrong-spelling-in-comment
+        # step 1: Montecarlo sampling
+        grads, output = self.run_montecarlo()
+
+        # step 2: compute as many fisher info. matrices as (input, params) sets
+        fishers = self.get_fishers(gradients=grads, model_outputs=output)
+
+        # step 3: get normalized fisher info matrices
+        f_hat, _ = self.get_fhat(fishers)
+
+        # step 4: compute eff. dim
+        effective_dims = self._get_eff_dim(f_hat, n)
+
+        return effective_dims
+
+
+class LocalEffectiveDimension(EffectiveDimension):
+    """
+    Computes the LOCAL effective dimension for a parametrized model.
+    """
+
+    def __init__(
+        self,
+        qnn: NeuralNetwork,
+        num_inputs: Optional[int] = 1,
+        params: Optional[Union[List, np.ndarray]] = None,
+        inputs: Optional[Union[List, np.ndarray]] = None,
+        fix_seed: bool = False,
+        callback: Optional[Callable] = None,
```

Comment:
Similar comment to the parent class in terms of some of the params, like num_inputs, callback and seed.

Here params just says List rather than List[float] in the typehint.

The order of the parameters here is different than the parent. Yes it has one less but for the others would it not make more sense to keep the same order?

Reply:
Yes, it would make more sense. Thanks for pointing it out!